src/Tribe/Cache/Abstract_Cache.php

```diff
@@ -74,8 +74,29 @@ protected function fetch_posts_with_ticket_types( array $post_types = null ) {
 			}
 		}
 
+
 		$ids = $wpdb->get_col( $query );
+		$ids = is_array( $ids ) ? $ids : array();
+
+		if ( empty( $ids ) ) {
+			return $ids;
+		}
+
+		/**
+		 * The above will fetch posts based on the meta data regardless of the status of the post, however post
+		 * under the status of `trash` or `auto-draft` shouldn't be in the list.
+		 */
+		$query = $wpdb->prepare(
+			"SELECT DISTINCT(ID) 
+				FROM {$wpdb->posts}
+				WHERE post_status != 'auto-draft' 
```

Comment:
We should allow filtering of these post status and use `NOT IN` to allow easier filtering.